habitat-lab/habitat/tasks/rearrange/rearrange_sensors.py

```diff
@@ -1627,11 +1630,16 @@ def get_observation(self, observations, *args, **kwargs):
         colors = []
         ik_helper = self._sim.get_agent_data(self.agent_id).ik_helper
         for point in down_sampled_points_xyz:
-            reachable = ik_helper.is_reachable(point)
+            cur_ee_pos = self._sim.articulated_agent.ee_transform().translation
+            reachable = ik_helper.is_reachable(cur_ee_pos, point, self.ctrl_lim)
             # Green if reachable, red if not
             colors.append([0, 255, 0] if reachable else [255, 0, 0])
-            
-        # Project the points to the image and color the pixels with circles 
+
+        if 'ee_target' in kwargs:
+            np.append(down_sampled_points_xyz, kwargs['ee_target'])
+            colors.append([0, 255, 0])
+
+        # Project the points to the image and color the pixels with circles
```

Comment:
Is this the visualization of current ee position?

Reply:
Right. But up to now, I can't actually visualize it since spot head_camera can't see ee_target due to its robot design. And there are stills some bugs with fetch and stretch.